### PR TITLE
Use Case Survey Notice

### PR DESCRIPTION
**Pull request by wiersgallak**
Adding in-product notice to ask for users to provide feedback on how we are supporting DevOp Use cases.

### CODE REVIEW

File: notices.json [hunk 1]
```diff
@@ -1,4 +1,23 @@
 [
+  {
+    "id": "use_case_survey",
+    "conditions": {
+      "audience": "all",
+      "clientType": "all",
+      "instanceType": "all",
+      "repeatable": "true",
```
Comment (esethna) [inline notices.json:8]:
```suggestion
```

File: notices.json [hunk 2]
```diff
@@ -1,4 +1,23 @@
 [
+  {
+    "id": "use_case_survey",
+    "conditions": {
+      "audience": "all",
+      "clientType": "all",
+      "instanceType": "all",
```
Comment (esethna) [inline notices.json:7]:
```suggestion
      "instanceType": "all"
```

File: notices.json [hunk 3]
```diff
@@ -1,4 +1,23 @@
...
+      "repeatable": "true",
+      "displayDate": "> 2022-08-01T00:00:00Z <= 2022-09-30T00:00:00Z",
+    },
+    "localizedMessages": {
+      "en": {
+        "title": "We want your feedback!",
+        "description": "We are looking for feedback on how we are doing in supporting your team's collaboration throughout different phases of the DevOps lifecycle. Please help us with a quick 5 minutes survey.",
```
Comment (esethna) [inline notices.json:14]: Propose mentioning the chance to win airpods in this copy to drive more action.

**Review by esethna — CHANGES_REQUESTED** [review]

**Comment by esethna** [comment]
While I definitely see the value here, just want to give fair warning that we are probably going to get some backlash for this notice similar to the feedback we got about the cloud freemium notice (since we specifically state in the system console and docs that the purpose of the notices is for product changes and admin updates). It may prompt some admins to disable notices all together.

**Comment by wiersgallak** [comment]
@esethna What would you think of a permanent link to a page that offers multiple areas to provide feedback? We can chat in our 1:1 but here's an idea I am playing with: https://portal.productboard.com/mattermost/33-looking-for-input

File: notices.json (hunk 1 above)
Comment (wiersgallak) [reply notices.json]: Done

File: notices.json (hunk 3 above)
Comment (wiersgallak) [reply notices.json]: Included.

**Review by esethna — APPROVED** [review]

**Comment by wiersgallak** [comment]
@ogi-m Can you please review this one so we can get it published?  Thanks in advance!

**Comment by ogi-m** [comment]
Hi @wiersgallak, could you please submit a PR against the `release` branch? The `master` branch is used for testing, so I'll go ahead and merge this one.

**Comment by brycied00d** [comment]
Hey all,
This may not be the right place to offer *this* feedback, but I wanted to confirm what @esethna said (https://github.com/mattermost/notices/pull/292#issuecomment-1204045250).

I've been hearing from my users that they want these notices disabled if they're just going to be "spam". Tentatively, I'm leaving End Users Notices enabled on my company's instance and hoping that future notices will better align with the feature's description (being used for "hey, update your client" and "hey, check out new feature" for example).

Perhaps in the future, a "server admin approval" process would allow me to filter what makes it through to my end users so I can filter what's appropriate from what they really don't want to be bothered by. (But that's a feature request for a different repo)

**Comment by wiersgallak** [comment]
@brycied00d Thank you for this feedback. Our intent was to welcome end-users with a direct opportunity to help us improve the product, but I appreciate your position on using this notification system for communicating feature and product improvements. We will also take into consideration your "admin approval" process as a future enhancement to this system. 

From a learning perspective for our team, what would be some ways that we could extract more feedback directly from your end-users in a more defined way? The freeform feedback data we get from NPS bot is great, but we want to find opportunities to ask more specific questions in the future to see if we need to adjust our product strategies. Happy to set up a call or take your input directly on this thread. Thank you for initiating a conversation with us around this topic.

**Comment by brycied00d** [comment]
Hi @wiersgallak , I appreciate you reaching out.

I do understand Mattermost's need for end-user feedback, and the fine line that is walked between pushing an important (to MM) message vs annoying your customers and end-users.

Speaking for my coworkers, I think their frustration stemmed mostly from the nature in which it was presented - a big pop-up modal that appeared, blocking their path, when they were just trying to go about their job and get stuff done. This sort of in-your-face interaction makes sense for "important information [for the end-user]", such as "hey your desktop client is out of date!", and to a lesser extent "hey we've updated, here are the new features" -- the former seems objectively important to me, the latter is not so objectively important but still useful to the end user. 

Spit-balling some ideas besides the "admin approval" method...
* A means for the users themselves to opt-out of notices entirely.
* Classification of the notices into buckets (information, feedback, critical), and a corresponding administrator control to allow specific categories.
  * And/or a means for the users to opt-out of certain buckets.
* Either in place of, or in addition to an "end-user notice" soliciting feedback, an easy-to-find button in the user interface. Perhaps tucked into the "menu" (the 3x3 grid of dots in the top left, where "System Console" and "Download Apps" are currently linked). Or as a button at the bottom of the "team gutter" (list of team icons on the left column).
  * By extension, if there is a specific feedback campaign happening, these buttons may be decorated with a small popup or notification indication. This would allow the user to notice them, and have their attention drawn even, without interrupting the user when they're busy. 
  * For instance, I'm mildly annoyed that after I dismissed that notice, I have "no way" (no obvious way, though I've since learned I could dig the URL out of the notices.mattermost.com payload) to go back and offer my feedback at a now-convenient time. A button in some fashion could lead me to re-open that notice later when I do have time.

To reiterate, my and my teams' main grievance was simply how disruptive this was. It was "spam" because it was not important to the user, and it made Mattermost look bad because it was shoved in their face.

Thank you again for your time. And kudos, you managed to extract feedback from me and my company's users after all :laughing:

**Comment by wiersgallak** [comment]
@brycied00d This is so helpful and appreciated! We never want to create an intentionally poor experience for users, so understanding what is frustrating and what other alternatives would create a better experience is crucial to us. 

I think there are some great ideas here that we can implement in future research initiatives.  I especially think having an easy-to-find button in the user interface could be welcoming, yet non-invasive method to get input. We are considering offering a portal with multiple ways to provide input  - surveys, designs, feature requests, and feature ideas we have that we want to validate/gauge interest in. Your idea about a small notification of new campaigns would be a great addition to that idea. 

Again, I've enjoyed hearing your input, and really do thank you for your time in giving me some deeper thoughts.

**Pull request merged**